compiler/packages/babel-plugin-react-compiler/src/Optimization/OutlineFunctions.ts

```diff
@@ -7,6 +7,36 @@
 
 import {HIRFunction, IdentifierId} from '../HIR';
 
+/**
+ * Checks if a function has any references to variables from outer scopes
+ * (beyond the explicitly tracked context array). This includes LoadContext
+ * instructions which indicate the function is accessing variables that are
+ * not passed as parameters or declared locally.
+ */
+function hasUnaccountedOuterScopeReferences(fn: HIRFunction): boolean {
```

Comment:
Maybe “accessesOuterContextVariables()” or similar. They’re accounted for, it’s a question of whether such variables are accessed or not.